src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -216,6 +218,31 @@ private void DetermineSerializationCapabilities()
             }
         }
 
+        /// <summary>
+        /// Return the JsonPropertyInfo for the TValue in IDictionary{string, TValue} when deserializing.
+        /// </summary>
+        /// <remarks>
+        /// This should not be called during warm-up (initial creation of JsonPropertyInfos) to avoid recursive behavior
+        /// which could result in a StackOverflowException.
+        /// </remarks>
+        public JsonPropertyInfo DictionaryValuePropertyPolicy
+        {
+            get
+            {
+                if (_dictionaryValuePropertyPolicy == null)
+                {
+                    Debug.Assert(ClassType == ClassType.Dictionary || ClassType == ClassType.IDictionaryConstructible);
+
+                    Type dictionaryValueType = ElementType;
+                    Debug.Assert(ElementType != null);
+
+                    _dictionaryValuePropertyPolicy = JsonClassInfo.CreateRootProperty(dictionaryValueType, Options);
```

Comment:
Should we first check `JsonClassInfo.ElementClassInfo.PolicyProperty` for this value in case it's already cached, then fallback on the call to `CreateRootProperty` if null is returned (which should only be for `ClassType.Object` element types)?

Reply:
Yes thanks.